crates/spicepod/src/component.rs

```diff
@@ -52,14 +55,61 @@ pub struct ComponentReference {
     pub depends_on: Vec<String>,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Serialize)]
 #[cfg_attr(feature = "schemars", derive(JsonSchema))]
 #[serde(untagged)]
 pub enum ComponentOrReference<T> {
     Component(T),
     Reference(ComponentReference),
 }
 
+impl<'de, T> Deserialize<'de> for ComponentOrReference<T>
+where
+    T: Deserialize<'de> + Debug,
+{
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct ComponentOrReferenceVisitor<T>(PhantomData<T>);
+
+        impl<'de, T> Visitor<'de> for ComponentOrReferenceVisitor<T>
```

Comment:
For learning purpose: Why is the implementation of Visitor inside the Deserializer implementation, instead of being outside